src/multipatch/connectivity/edge_transformation.hpp

```diff
@@ -223,29 +259,58 @@ class EdgeTransformation
                 !std::is_same_v<TargetGrid, CurrentGrid>,
                 "The types of the indices should be different");
 
+        using TargetPatch
+                = std::conditional_t<std::is_same_v<TargetGrid, EdgeGrid1>, Patch1, Patch2>;
+
         // Index range of CurrentIdx.
         using IdxRangeCurrent = IdxRange<CurrentGrid>;
         // Index range of other dimension.
         using IdxRangeTarget = IdxRange<TargetGrid>;
         // Coordinate on the discontinuous dimension of the current grid.
         using CurrentCoord = Coord<typename CurrentGrid::continuous_dimension_type>;
-        // Index on the target grid
-        using IdxTarget = typename IdxRangeTarget::discrete_element_type;
         // Index step on the target grid
         using IdxStepTarget = typename IdxRangeTarget::discrete_vector_type;
 
+        /*
+            Add a point for the periodic case where the last interpolation point
+            is usually not on the last break point. 
+        */
+        IdxRangeEdge1 const idx_range_patch_1_full(
+                m_idx_range_patch_1.front(),
+                m_idx_range_patch_1.extents() + int(EdgeDim1::PERIODIC));
+        IdxRangeEdge2 const idx_range_patch_2_full(
+                m_idx_range_patch_2.front(),
+                m_idx_range_patch_2.extents() + int(EdgeDim2::PERIODIC));
+
 
         bool is_equivalent_idx = false;
 
         // Get the 1D index range corresponding to the current and target domains.
         IdxRange<EdgeGrid1, EdgeGrid2> const
-                combined_idx_range(m_idx_range_patch_1, m_idx_range_patch_2);
+                combined_idx_range(idx_range_patch_1_full, idx_range_patch_2_full);
         IdxRangeCurrent const current_idx_range(combined_idx_range);
         IdxRangeTarget const target_idx_range(combined_idx_range);
 
         int const n_cells_current = current_idx_range.size() - 1;
         int const n_cells_target = target_idx_range.size() - 1;
 
+        // Periodicity property.
+        if constexpr (CurrentGrid::continuous_dimension_type::PERIODIC) {
+            std::size_t counter = 0;
+            while ((current_idx < current_idx_range.front()) && (counter < 10)) {
+                counter++;
+                current_idx += n_cells_current;
+            }
+            counter = 0;
+            while ((current_idx >= current_idx_range.back()) && (counter < 10)) {
+                counter++;
+                current_idx -= n_cells_current;
+            }
+        } else {
+            assert(current_idx <= current_idx_range.back());
+            assert(current_idx >= current_idx_range.front());
+        }
```

Comment:
```suggestion
        }
        assert(current_idx <= current_idx_range.back());
        assert(current_idx >= current_idx_range.front());
```
If you want to be completely general I guess these assertions shouldn't be in the `else` condition as you have no guarantee that the periodicity returns to the correct range (due to the n_iter_max = 10)